scripts/testnet.sh

```diff
@@ -46,18 +49,84 @@ get_rpc() {
   echo "${unresolved_rpc}" | envsubst
 }
 
+# $1 = environment
+# $2 = token (native | hopr)
+funding_wallet_info() {
+  local environment="${1}"
+  local token="${2}"
+  curl --silent "$API_ENDPOINT/api/faucet/$environment/info?text=$token"
+}
+
+# $1 = environment
+# $2 = address
+# $3 = token
+wallet_balance() {
+  local environment="${1}"
+  local address="${2}"
+  curl --silent "$API_ENDPOINT/api/balance/$environment/$address/$token?text=true"
+}
+
+# $1 = environment
+# $2 = address
+# $3 = token (native | hopr)
+faucet_to_address() {
+  local environment="${1}"
+  local address="${2}"
+  local token="${3}"
+  local secret="${FAUCET_SECRET_API_KEY}"
+
+  curl --silent --request POST \
+  "$API_ENDPOINT/api/faucet/$environment/$address/$token?text=true" \
+  --header 'Content-Type: application/json' \
+  --data-raw "{\"secret\": \"$secret\"}"
+}
+
 # $1=account (hex)
 # $2=environment
 fund_if_empty() {
   local address="${1}"
   local environment="${2}"
 
-  # start funding in sequence (nonce-tracker will not be able to keep track of nonce in two processes)
-  # we need to use yarn explicitely to ensure packages can be resolved properly
-  local fundNativeCmd="yarn --silent run ts-node ${mydir}/fund-address.ts \
-	  --environment ${environment} --address ${address} --targetERC20 ${min_funds_hopr} --targetNative ${min_funds}"
-  # wait 15 seconds in between retries to prevent race conditions
-  PRIVATE_KEY=${FUNDING_PRIV_KEY} try_cmd "${fundNativeCmd}" 3 15
+  local faucet_address
```

Comment:
Given that the API is environment aware I would like to see all of this functionality (checks, dual funding) in the API to reduce script footprint.

Reply:
True, this could be handled in the API, but thought visibility here would make more sense as we would like to retry if things don't work out and the API has no way to expose this.

I believe this is a good place to start and we can migrate any functionality later. I rather not be picky, as we have yet to verify this works, which we should test first before going into the semantics of where we should put the `if` statements.